lib/fizzy/parser.cpp

```diff
@@ -142,9 +143,11 @@ inline parser_result<std::string> parse_string(const uint8_t* pos, const uint8_t
     std::vector<uint8_t> value;
     std::tie(value, pos) = parse_vec<uint8_t>(pos, end);
 
-    // FIXME: need to validate that string is a valid UTF-8
+    const auto ret = std::string(value.begin(), value.end());
+    if (!utf8_validate(ret))
+        throw parser_error{"Invalid UTF8"};
 
-    return {std::string(value.begin(), value.end()), pos};
+    return {ret, pos};
```

Comment:
```suggestion
    return {std::move(ret), pos};
```

Reply:
clang-tidy complained that moving `const string` is useless.

Reply:
yes, it is, you should make it non-const first

Reply:
Yeah, but will it not be moved automatically if i is const? That's the intuition.

Reply:
on the contrary, if it's bound to const variable, this variable can't be modified, so it can't be moved out of this variable

Reply:
Okay, fixed.